src/NServiceBus.Persistence.DynamoDB.Tests/InstallerTests.cs

```diff
@@ -2,155 +2,229 @@
-            Assert.AreEqual(outboxSettings.SortKeyName, table.Table.KeySchema[1].AttributeName);
-            Assert.AreEqual(KeyType.RANGE, table.Table.KeySchema[1].KeyType);
-            Assert.AreEqual(ScalarAttributeType.S, table.Table.AttributeDefinitions.Single(a => a.AttributeName == outboxSettings.SortKeyName).AttributeType);
-            Assert.AreEqual(TableStatus.ACTIVE, table.Table.TableStatus);
-        }
+                Assert.AreEqual(BillingMode.PAY_PER_REQUEST, table.Table.BillingModeSummary.BillingMode);
+                Assert.AreEqual(0, table.Table.ProvisionedThroughput.ReadCapacityUnits);
+                Assert.AreEqual(0, table.Table.ProvisionedThroughput.WriteCapacityUnits);
+            }
 
-        [Test]
-        public async Task Should_create_outbox_with_pay_as_you_go_billing_mode()
-        {
-            installerSettings.BillingMode = BillingMode.PAY_PER_REQUEST;
+            [Test]
+            public async Task Should_create_table_with_provisioned_billing_mode()
+            {
+                sagaSettings.ProvisionedThroughput = new ProvisionedThroughput(1, 1);
+                sagaSettings.BillingMode = BillingMode.PROVISIONED;
 
-            await installer.Install();
+                await installer.CreateSagaTableIfNotExists(sagaSettings, CancellationToken.None);
 
-            var table = await dynamoClient.DescribeTableAsync(outboxSettings.TableName);
+                var table = await dynamoClient.DescribeTableAsync(sagaSettings.TableName);
 
-            Assert.AreEqual(BillingMode.PAY_PER_REQUEST, table.Table.BillingModeSummary.BillingMode);
-            Assert.AreEqual(0, table.Table.ProvisionedThroughput.ReadCapacityUnits);
-            Assert.AreEqual(0, table.Table.ProvisionedThroughput.WriteCapacityUnits);
-        }
+                // Don't assert on BillingModeSummary as it may be not set when using provisioned mode.
+                Assert.AreEqual(sagaSettings.ProvisionedThroughput.ReadCapacityUnits, table.Table.ProvisionedThroughput.ReadCapacityUnits);
+                Assert.AreEqual(sagaSettings.ProvisionedThroughput.WriteCapacityUnits, table.Table.ProvisionedThroughput.WriteCapacityUnits);
+            }
 
-        [Test]
-        public async Task Should_create_outbox_with_provisioned_billing_mode()
-        {
-            installerSettings.ProvisionedThroughput = new ProvisionedThroughput(1, 1);
-            installerSettings.BillingMode = BillingMode.PROVISIONED;
+            [Test]
+            public async Task Should_not_throw_when_same_table_already_exists()
+            {
+                sagaSettings.PartitionKeyName = "PK1";
+                await installer.CreateSagaTableIfNotExists(sagaSettings, CancellationToken.None);
 
-            await installer.Install();
+                var table1 = await dynamoClient.DescribeTableAsync(sagaSettings.TableName);
```

Comment:
Should there be some kind of delay loop here to avoid consistency-related failures (the table "doesn't exist yet" when calling Describe for example)

Reply:
I'm not aware of describe returning inconsistent data tbh. Do you have any reference to that?

Reply:
(also the installer is already containing delays to only return once the table is active)